### DISCUSSION TITLE
Cost Optimization

### DISCUSSION THREAD

**Discussion by syndbg** [external-secrets/kubernetes-external-secrets#793] [General]
Hey, community of external-secrets,

We're looking to integrate external-secrets with AWS Secrets Manager as our preferred way to utilize secrets (in k8s).

We've run a few numbers and noticed that the cost for integrating external-secrets out of the door with 10k secrets is ~$2232/month with 10k secrets.

This is given that we enable polling  = 1m. 

Anyone else that has considered optimizing the cost? What "tricks" did you use :)

**Comment by Flydiverny** [comment]
If this is a new setup I would recommend evaluating the go rewrite ESO https://github.com/external-secrets/external-secrets ! Though in regards to polling 10k secrets every minute it would probably come out the same cost wise.
Otherwise for tricks, I'd say longer poll time is the main cost saver 😄 
As mentioned in other places the ideal way would be to be notified by whatever secret backend that a secret has changed and then re-sync those effected.
